File: python/LeetCode2582/main.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        t = time % (n - 1)
        return n - t if time // (n - 1) % 2 else t + 1

s1 = Solution()
print(s1.passThePillow(4, 5))

s2 = Solution()
print(s2.passThePillow(3, 2))

File: cpp/LeetCode2582/main.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    int passThePillow(int n, int time) {
        int t = time % (n - 1);
        return time / (n - 1) % 2 ? n - t : 1 + t;
    }
};

int main() {
    Solution s0;
    cout << s0.passThePillow(4, 5) << endl;
    Solution s1;
    cout << s1.passThePillow(3, 2) << endl;
    return 0;
}
